src/irr/asset/CCPUMeshPacker.h

```diff
@@ -295,99 +287,266 @@ uint32_t CCPUMeshPacker<MDIStructType>::processMeshBuffer(ICPUMeshBuffer* inputM
+				{
+					const uint32_t oldIndex = triangle.oldIndices[j];
+					auto result = usedVertices.insert(std::make_pair(oldIndex, newIdx));
+
+					newIdxTris[i].oldIndices[j] = result.second ? newIdx++ : result.first->second;			
+				}
 			}
-			else
+			
+			//TODO: cache optimization
+
+			//copy indices into unified index buffer
+			for (size_t i = 0; i < batch.triangles.size(); i++)
 			{
+				for (int j = 0; j < 3; j++)
+				{
+					*indexBuffPtr = newIdxTris[i].oldIndices[j];
+					indexBuffPtr++;
+				}
+			}
+
+			//copy deinterleaved vertices into unified vertex buffer
+
+			for (uint16_t attrBit = 0x0001, location = 0; location < SVertexInputParams::MAX_ATTR_BUF_BINDING_COUNT; attrBit <<= 1, location++)
+			{
+				if (!(m_outVtxInputParams.enabledAttribFlags & attrBit))
+					continue;
+
+				SVertexInputAttribParams attrib = m_outVtxInputParams.attributes[location];
 				SVertexInputAttribParams MBAttrib = (*it)->getPipeline()->getVertexInputParams().attributes[location];
+
 				SVertexInputBindingParams attribBinding = (*it)->getPipeline()->getVertexInputParams().bindings[MBAttrib.binding];
 				uint8_t* attrPtr = (*it)->getAttribPointer(location);
+				const size_t attrSize = asset::getTexelOrBlockBytesize(static_cast<E_FORMAT>(attrib.format));
 				const size_t stride = (attribBinding.stride) == 0 ? attrSize : attribBinding.stride;
 
-				switch (attribBinding.inputRate)
-				{
-				case EVIR_PER_VERTEX:
-				{
-					for (uint64_t i = 0; i < vtxCnt; i++)
-					{
-						//assert((perVertexBuffPtr + attrSize) <= ((ramb.vertexAllocationOffset * m_vtxSize) + ramb.vertexAllocationReservedSize));
-						memcpy(vtxBuffPtr, attrPtr, attrSize);
-						vtxBuffPtr += attrSize;
-						attrPtr += stride;
-					}
-					break;
-				}
-				case EVIR_PER_INSTANCE:
+				SBufferBinding<ICPUBuffer>& vtxBuffBind = outputBuffer.vertexBufferBindings[location];
+				uint8_t* outBuffAttrPtr = static_cast<uint8_t*>(vtxBuffBind.buffer->getPointer()) + vtxBuffBind.offset;
+				outBuffAttrPtr += (ramb.vertexAllocationOffset + verticesAddedToUnifiedBufferCnt) * attrSize;
+
+				//if (location == 0)
+				//{
+				//	for (auto index : usedVertices)
+				//	{
+				//		std::cout << '\n' << index.first << ' ' << index.second << std::endl;
+				//	}
+
+				//	std::cout << "-------------\n";
+				//}
+				
+				for (auto index : usedVertices)
 				{
-					//not implemented yet
-					_IRR_DEBUG_BREAK_IF(true);
-					assert(false);
-					assert(m_perInstVtxSize);
-
-					for (uint64_t i = 0; i < /*fix*/vtxCnt; i++)
-					{
-						//assert((perInstBuffPtr + attrSize) <= ((ramb.instanceAllocationOffset * m_perInstVtxSize) + ramb.instanceAllocationReservedSize));
-						memcpy(vtxBuffPtr, attrPtr, attrSize);
-						vtxBuffPtr += attrSize;
-						attrPtr += stride;
-					}
-					break;
-				}
+					const uint8_t* attrSrc = attrPtr + (index.first * stride);
+					uint8_t* vtxAttrDest = outBuffAttrPtr + (index.second * attrSize);
+					memcpy(vtxAttrDest, attrSrc, attrSize);	
```

Comment:
I'd still like to see you use the `ICPUMeshBuffer` static methods to advance and compute pointers if possible